src/framework/mlt_image.c

```diff
@@ -425,3 +426,215 @@ void mlt_image_format_planes( mlt_image_format format, int width, int height, vo
 		strides[3] = 0;
 	};
 }
+
+typedef struct
+{
+	mlt_image src;
+	mlt_image dst;
+	int radius;
+} blur_slice_desc;
+
+static int blur_h_proc(int id, int index, int jobs, void* data)
```

Comment:
Image processing functions do not belong in the framework.

Reply:
I can accept that. The idea was inspired by FFmpeg which has a gaussian function in swcale. It is part of the public interface, and multiple filters use it.
https://github.com/FFmpeg/FFmpeg/blob/069f7831a2bd1545080121fb6d0773191cd61d70/libswscale/swscale.h#L351

Maybe someday there could be a module in the framework for useful processing functions. I will make a separate proposal if I come up with something.

Reply:
This idea is a little complicated within MLT proper. There could be a libmltimgproc library sub-project added that would be a better fit. But then, is there not already enough image processing libraries? And why go into this trouble when all image processing must be completely revisited for higher than 8-bit processing?